src/NewTriggerDeliveryPolicyPage.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+This file is part of Astarte.
+
+Copyright 2023 SECO Mind Srl
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/* eslint-disable camelcase */
+
+import React, { useCallback, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Button, Container, Row, Spinner } from 'react-bootstrap';
+
+import { AlertsBanner, useAlerts } from './AlertManager';
+import { useAstarte } from './AstarteManager';
+import BackButton from './ui/BackButton';
+import TriggerDeliveryPolicyEditor from './components/TriggerDeliveryPolicyEditor';
+import {
+  AstarteTriggerDeliveryPolicyDTO,
+  AstarteTriggerDeliveryPolicyHandlerDTO,
+} from './astarte-client/types/dto/triggerDeliveryPolicy';
+
+export default (): React.ReactElement => {
+  const [policyDraft, setPolicyDraft] = useState<AstarteTriggerDeliveryPolicyDTO>();
+  const [isValidPolicy, setIsValidPolicy] = useState(false);
+  const [isInstallingPolicy, setIsInstallingPolicy] = useState(false);
+  const [isSourceVisible, setIsSourceVisible] = useState(true);
+  const [installationAlerts, installationAlertsController] = useAlerts();
+  const astarte = useAstarte();
+  const navigate = useNavigate();
+  let newHandlers: AstarteTriggerDeliveryPolicyHandlerDTO[] = [];
+
+  const handleToggleSourceVisibility = useCallback(() => {
+    setIsSourceVisible((isVisible) => !isVisible);
+  }, []);
+
+  const handlePolicyChange = useCallback(
+    (updatedPolicy: AstarteTriggerDeliveryPolicyDTO, isValid: boolean) => {
+      setPolicyDraft(updatedPolicy);
+      setIsValidPolicy(isValid);
+    },
+    [],
+  );
+
+  const validatePolicy = (policy?: AstarteTriggerDeliveryPolicyDTO) => {
+    policy?.error_handlers.forEach((el, index) => {
+      const str = el.on.toString();
+      const array = Array.from(str.split(','), Number);
+      const newArray = array.filter((v, i) => array.indexOf(v) === i);
```

Comment:
It's a good practice to name variables with what they are supposed to contain, rather than their type.

```javascript
const errorHandlerCodes = el.on.toString();
const codes = ...
```